src/context/CategoryContext.jsx

```diff
@@ -0,0 +1,32 @@
+import axios from "axios";
+import React, { createContext, useContext, useEffect, useState } from "react";
+
+const categoryContext = createContext([]);
+
+const useCategory = () => useContext(categoryContext);
+
+function CategoryProvider({ children }) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/categories");
+
+        if (response.status === 200) {
+          setCategories(response.data.categories);
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    })();
+  }, []);
```

Comment:
there is no need to give empty array dependency here as context will run only once anyways